projects/forms/src/lib/validators/has-no-future-date/has-no-future-date.validator.ts

```diff
@@ -0,0 +1,30 @@
+import { FormControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { isValid } from 'date-fns';
+
+/**
+ * hasNoFutureDateValidator
+ *
+ * Validator function to ensure that the selected date is not in the future.
+ * If the date is in the future, it returns an error.
+ * @returns ValidationErrors if the date is in the future, otherwise null.
+ *
+ */
+export const hasNoFutureDateValidator = (): ValidatorFn => {
+	return (control: FormControl): ValidationErrors | null => {
+		// Early exit in case the control or the value does not exist
+		if (!control.value) {
+			return null;
+		}
+
+		// Create date objects based on the provided date and current date
+		const inputDate = new Date(control.value);
+		const currentDate = new Date();
+
+		// In case the date itself is invalid, we early exit to let a potential date validator handle the error
```

Comment:
Ik heb het al geapproved want dit is een micro performantie upgrade; maar je kan deze ook boven de `const currentDate = new Date()` functie zetten; dan early exit de validator al vooraleer je nog een nieuwe date aanmaakt. 

Again, kleine change en minimale performantie upgrade, maar het zijn vaak van die kleintjes die het doen!